From 29e426807fe814a4f697ed844bdd052abd7a9430 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Tue, 15 Feb 2022 18:11:09 +0000
Subject: [PATCH] AppPackage: allow different owner UIDs/GIDs

By default, when tar is started as root, it tries to keep the same
ownership ids as in the archive. It that fails, then tar exits with
an error.

On the HP Touchpad, the extraction is done in /media/cryptofs, which
is a vfat filesystem, and that doesn't handle UIDs/GIDs.
But more generally, the installer shouldn't trust the package to have
always sane ownership values, and should impose its own values.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 src/installer/AppPackage.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/installer/AppPackage.cpp b/src/installer/AppPackage.cpp
index 76a7dfe..27bc88f 100644
--- a/src/installer/AppPackage.cpp
+++ b/src/installer/AppPackage.cpp
@@ -152,6 +152,7 @@ bool AppPackage::extractOneItem()
     argv[index++] = (gchar *) "tar";
     argv[index++] = (gchar *) "xzf";
     argv[index++] = (gchar *) targetFile.c_str();
+    argv[index++] = (gchar *) "--no-same-owner";
     argv[index] = NULL;
 
     result = g_spawn_async(m_targetPath.c_str(),
